File: source/aoImportWizard/Views/SelectSourceView.vb

Imports Contensive.BaseClasses
Imports Contensive.ImportWizard.Models

Namespace Contensive.ImportWizard.Controllers
    Public Class SelectSourceView
        '
        ''' <summary>
        ''' return the next view. 0 goes to the first form (start over)
        ''' </summary>
        ''' <param name="app"></param>
        ''' <returns></returns>
        Public Shared Function processView(app As ApplicationModel, srcViewId As Integer) As Integer
            Try
                Dim cp As CPBaseClass = app.cp
                Dim Button As String = app.cp.Doc.GetText(RequestNameButton)
                If String.IsNullOrEmpty(Button) Then Return srcViewId
                If Button = ButtonCancel Then
                    '
                    ' Cancel
                    ImportConfigModel.clear(app)
                    Return viewIdReturnBlank
                End If
                If Button = ButtonRestart Then
                    '
                    ' Restart
                    ImportConfigModel.clear(app)
                    Return viewIdSelectSource
                End If
                '
                Dim importConfig As ImportConfigModel = ImportConfigModel.create(app)
                importConfig.importSource = CType(app.cp.Doc.GetInteger(RequestNameImportSource), ImportDataModel_ImportTypeEnum)
                importConfig.save(app)
                '
                Select Case Button
                    Case ButtonBack
                        '
                        Return srcViewId
                    Case ButtonContinue
                        '
                        Select Case importConfig.importSource
                            Case ImportDataModel_ImportTypeEnum.UploadFile
                                '
                                ' -- upload a commad delimited file
                                Return viewIdUpload
                            Case Else
                                '
                                ' -- use a file uploaded previously
                                Return viewIdSelectFile
                        End Select
                End Select
            Catch ex As Exception
                app.cp.Site.ErrorReport(ex)
                Throw
            End Try
        End Function
        '
        ''' <summary>
        ''' return the html for this view
        ''' </summary>
        ''' <param name="app"></param>
        ''' <returns></returns>
        Public Shared Function getView(app As ApplicationModel) As String
            Try
                Dim cp As CPBaseClass = app.cp
                Dim importConfig As ImportConfigModel = ImportConfigModel.create(app)
                Dim headerCaption As String = "Import Wizard"
                Dim importSource As Integer = Convert.ToInt32(importConfig.importSource)
                Dim description As String = cp.Html.h4("Select the import source") & cp.Html.p("If you want to upload a new comma-delimited text file (csv), select the upload option. If you want to use a file you previously uploaded, select the second option.")
                Dim content As String = "" _
                    & "<div class=""ml-4"">" _
                    & "<div class=""pb-2"">" _
                    & "<label>" & cp.Html.RadioBox(RequestNameImportSource, ImportSourceUpload.ToString, importSource.ToString, "mr-2") & "Upload a comma delimited text file (up to 5 MBytes).</label>" _
                    & "</div>" _
                    & "<div class=""pb-2"">" _
                    & "<label>" & cp.Html.RadioBox(RequestNameImportSource, ImportSourceUploadFolder.ToString, importSource.ToString, "mr-2") & "Use a file you previously uploaded.</label>" _
                    & "</div>" _
                    & "</div>" _
                    & ""
                content &= cp.Html5.Hidden(rnSrcViewId, viewIdSelectSource)
                Return HtmlController.createLayout(cp, headerCaption, description, content, True, False, False, True)
            Catch ex As Exception
                app.cp.Site.ErrorReport(ex)
                Throw
            End Try
        End Function
    End Class
End Namespace

File: source/aoImportWizard/Views/SelectSourceView.cs
using System;
using Contensive.ImportWizard.Models;

namespace Contensive.ImportWizard.Controllers {
    public class SelectSourceView {
        // 
        /// <summary>
        /// return the next view. 0 goes to the first form (start over)
        /// </summary>
        /// <param name="app"></param>
        /// <returns></returns>
        public static int processView(ApplicationModel app, int srcViewId) {
            try {
                var cp = app.cp;
                string Button = app.cp.Doc.GetText(constants.RequestNameButton);
                if (string.IsNullOrEmpty(Button))
                    return srcViewId;
                if ((Button ?? "") == constants.ButtonCancel) {
                    // 
                    // Cancel
                    ImportConfigModel.clear(app);
                    return constants.viewIdReturnBlank;
                }
                if ((Button ?? "") == constants.ButtonRestart) {
                    // 
                    // Restart
                    ImportConfigModel.clear(app);
                    return constants.viewIdSelectSource;
                }
                // 
                var importConfig = ImportConfigModel.create(app);
                importConfig.importSource = (ImportDataModel_ImportTypeEnum)app.cp.Doc.GetInteger(constants.RequestNameImportSource);
                importConfig.save(app);
                // 
                switch (Button ?? "") {
                    case constants.ButtonBack: {
                            // 
                            return srcViewId;
                        }
                    case constants.ButtonContinue: {
                            // 
                            switch (importConfig.importSource) {
                                case ImportDataModel_ImportTypeEnum.UploadFile: {
                                        // 
                                        // -- upload a commad delimited file
                                        return constants.viewIdUpload;
                                    }

                                default: {
                                        // 
                                        // -- use a file uploaded previously
                                        return constants.viewIdSelectFile;
                                    }
                            }

                            break;
                        }
                }
            } catch (Exception ex) {
                app.cp.Site.ErrorReport(ex);
                throw;
            }

            return default;
        }
        // 
        /// <summary>
        /// return the html for this view
        /// </summary>
        /// <param name="app"></param>
        /// <returns></returns>
        public static string getView(ApplicationModel app) {
            try {
                var cp = app.cp;
                var importConfig = ImportConfigModel.create(app);
                string headerCaption = "Import Wizard";
                int importSource = Convert.ToInt32((int)importConfig.importSource);
                string description = cp.Html.h4("Select the import source") + cp.Html.p("If you want to upload a new comma-delimited text file (csv), select the upload option. If you want to use a file you previously uploaded, select the second option.");
                string content = "" + "<div class=\"ml-4\">" 
                    + "<div class=\"pb-2\">" 
                    + "<label>" + cp.Html.RadioBox(constants.RequestNameImportSource, constants.ImportSourceUpload.ToString(), importSource.ToString(), "mr-2") + "&nbsp;Upload a comma delimited text file (up to 5 MBytes).</label>" 
                    + "</div>" 
                    + "<div class=\"pb-2\">" + "<label>" + cp.Html.RadioBox(constants.RequestNameImportSource, constants.ImportSourceUploadFolder.ToString(), importSource.ToString(), "mr-2") + "&nbsp;Use a file you previously uploaded.</label>" + "</div>" + "</div>" + "";








                content += cp.Html5.Hidden(constants.rnSrcViewId, constants.viewIdSelectSource);
                return HtmlController.createLayout(cp, headerCaption, description, content, true, false, false, true);
            } catch (Exception ex) {
                app.cp.Site.ErrorReport(ex);
                throw;
            }
        }
    }
}